src/ripple/overlay/impl/PeerImp.h

```diff
@@ -339,7 +339,7 @@ class PeerImp : public Peer,
     void
     checkTracking(std::uint32_t seq1, std::uint32_t seq2);
 
-    PublicKey const&
+    PublicKey
     getNodePublic() const override
     {
         return publicKey_;
```

Comment:
I really don't like this. It looks like you got rid of the `const&` because you need to generate a `PublicKey` on the fly in the `TestPeer` test class. Instead of forcing an unnecessary copy to accommodate a test, change the test class to store that generated key in a class member variable, then return the variable.